src/script/cpp_api/s_entity.cpp

```diff
@@ -198,14 +226,70 @@ void ScriptApiEntity::luaentity_Step(u16 id, float dtime,
 	luaL_checktype(L, -1, LUA_TFUNCTION);
 	lua_pushvalue(L, object); // self
 	lua_pushnumber(L, dtime); // dtime
-	/* moveresult */
-	if (moveresult)
-		push_collision_move_result(L, *moveresult);
-	else
+
+	pushFloatPos(L, pos); // pos
+	push_v3f(L, rotation * core::DEGTORAD); // rotation (in radians)
```

Comment:
I wouldn't worry about breaking compatibility since the current API is not yet marked as stable. I think it is more important to have a logical order of parameters before the final release of 5.3.0. And in this case there is universal utility in knowing the position and velocity changes of all objects, whereas only physical objects can experience collisions.